packages/apollo-engine-reporting/src/treeBuilder.ts

```diff
@@ -233,14 +228,19 @@ function durationHrTimeToNanos(hrtime: [number, number]) {
 
 // Convert from the linked-list ResponsePath format to a dot-joined
 // string. Includes the full path (field names and array indices).
-function responsePathAsString(p: ResponsePath | undefined) {
+function responsePathAsString(p: ResponsePath | undefined): string {
```

Comment:
I realize you didn't introduce it originally, but would you support making `p` optional and dropping the `| undefined`?

```suggestion
function responsePathAsString(p?: ResponsePath): string {
```

(and also leaving out the explicit `undefined` at the call site of this function on line 19, as I noted in a separate GH suggestion?)